src/core/qgsmaplayerfactory.h

```diff
@@ -73,6 +73,12 @@ class CORE_EXPORT QgsMapLayerFactory
 
       //! Set to TRUE if the default layer style should be loaded
       bool loadDefaultStyle = true;
+
+      /**
+       *  Set to TRUE if the statistics calculation should be skipped
+       *  \since QGIS 3.26
+       */
+      bool skipStatisticsCalculation = false;
```

Comment:
Also this one, the options here are only meant for all sub-types of map layers...